### PR TITLE
Fix the problem that atlas will create an extra one

### PR DESCRIPTION
**Pull request by qiuguohua**
Re: #

### Changelog

https://github.com/cocos/cocos-engine/issues/16797#event-12228949916

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/utils/dynamic-atlas/atlas-manager.ts [hunk 1]
```diff
@@ -145,7 +145,7 @@ export class DynamicAtlasManager extends System {
 
     private newAtlas (): Atlas {
         let atlas = this._atlases[++this._atlasIndex];
-        if (!atlas) {
+        if (!atlas && this._atlasIndex < this.maxAtlasCount) {
```
Comment (minggo) [inline cocos/2d/utils/dynamic-atlas/atlas-manager.ts:149]: If so, then this function may return null. Should change the return type.

File: cocos/2d/utils/dynamic-atlas/atlas-manager.ts [hunk 2]
```diff
@@ -179,7 +179,7 @@ export class DynamicAtlasManager extends System {
         texture: DynamicAtlasTexture;
     } | null {
         if (EDITOR_NOT_IN_PREVIEW) return null;
-        if (!this._enabled || this._atlasIndex === this._maxAtlasCount
+        if (!this._enabled || this._atlasIndex >= this._maxAtlasCount
```
Comment (minggo) [inline cocos/2d/utils/dynamic-atlas/atlas-manager.ts:183]: When can this._atlasIndex greater than this._maxAtlasCount?

Comment (qiuguohua) [reply cocos/2d/utils/dynamic-atlas/atlas-manager.ts:183]: This might be safer

File: cocos/2d/utils/dynamic-atlas/atlas-manager.ts (hunk 1 above)
Comment (qiuguohua) [reply cocos/2d/utils/dynamic-atlas/atlas-manager.ts:149]: Yes, I'll change it

**Review by minggo — APPROVED** [review]

**Comment by minggo** [comment]
Please fix test error.

**Review by minggo — APPROVED** [review]

**Pull request merged**